hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java

```diff
@@ -331,6 +333,11 @@ private void checkNodeState(Operation op) {
         if (nodeEngine.getClusterService().getClusterState() == ClusterState.PASSIVE) {
             throw new IllegalStateException("Cluster is in " + ClusterState.PASSIVE + " state! Operation: " + op);
         }
+
+        InternalHotRestartService hotRestartService = node.getNodeExtension().getInternalHotRestartService();
+        if (hotRestartService.isEnabled() && !hotRestartService.isStartCompleted()) {
+            throw new RetryableHazelcastException("Recovery from persistence is still in progress. Operation: " + op);
```

Comment:
why don't we have this check in previous versions?

Reply:
Prior to 5.0, hot-restart recovery was only performed cluster-wide and in `PASSIVE` state, so there was no need for that. However, starting with 5.0, it is an omission (along with the change in `MapProxySupport#initialize` & pre-join ops fix to implement `AllowedDuringPassiveState`) and should be backported to `5.1.z` & `5.0.z`.

Reply:
should i open an issue for this?

Reply:
i have added a task for backports in the PR description